File: easy/slang_flavor.c
#include <stdbool.h>
#include <stdio.h>

static const char *phrases[] = { ", yeah!", ", this is crazy, I tell ya.",
				 ", can U believe this?", ", eh?", ", aw yea.",
				 ", yo.", "? No way!", ". Awesome!" };

int main(int argc, char *argv[]) {
	FILE *fp;
	char c;
	bool p = true, l = false;
	int i = 0;

	if (argc != 2) {
		printf("Usage: %s [FILE]\n", argv[0]);
		return 1;
	}
	fp = fopen(*++argv, "r");
	while ((c = getc(fp)) != EOF) {
		if (c == '.' || c == '!' || c == '?') {
			if (l) {
				printf("%s", phrases[i]);
				l = false;
				i = (i + 1) % 8;
			} else {
				putchar(c);
				l = true;
			}
		} else {
			putchar(c);
		}
		p = c == '\n';
	}
	if (!p)
		putchar('\n');
	return 0;
}

File: easy/slang_flavor.r
phrases <- c(", yeah!", ", this is crazy, I tell ya.", ", can U believe this?", ", eh?", ", aw yea.", ", yo.", "? No way!", ". Awesome!")
o <- 1
f <- FALSE

cat(sapply(readLines(tail(commandArgs(), n=1)), function(s) {
  t <- gregexpr(pattern='[.!?]', s)[[1]]
  l <- 1
  u <- ""
  for (i in 1:length(t)) {
    if (t[i] > 1) {
      u <- paste(u, substr(s, l, t[i] - 1), sep="")
    }
    if (f) {
      u <- paste(u, phrases[o], sep="")
      o <<- o %% 8 + 1
    } else {
      u <- paste(u, substr(s, t[i], t[i]), sep="")
    }
    l <- t[i] + 1
    f <<- !f
  }
  if (l < nchar(s)) {
    u <- paste(u, substr(s, l, nchar(s)), sep="")
  }
  u
}), sep="\n")
